src/test/test-macro.c

```diff
@@ -433,4 +433,27 @@ TEST(DECIMAL_STR_MAX) {
         assert_se(DECIMAL_STR_MAX(uint64_t) == DECIMAL_STR_WIDTH(u64_longest)+1);
 }
 
+TEST(PTR_SUB1) {
+        static const uint64_t x[4] = { 2, 3, 4, 5 };
```

Comment:
Also, any particular reason why not just `int`?

Reply:
i wanted something large (i.e. definitely not byte sized) that ensures we count correctly, i.e. take the pointer type size into account correctly. Now `int` are not byte sized either on our relevant archs, but it still felt better to make this explicit by picking a fixed size integer...